openmetadata-ui/src/main/resources/ui/src/components/PipelineStatusList/PipelineStatusList.component.tsx

```diff
@@ -92,11 +98,24 @@ const PipelineStatusList: FC<Prop> = ({
             </div>
           </div>
           {!isEmpty(executions) ? (
-            <ExecutionStrip
-              executions={executions as PipelineStatus}
-              selectedExecution={selectedExec}
-              onSelectExecution={onSelectExecution}
-            />
+            <Space size={2}>
+              {executions
+                .filter((execution) =>
+                  selectedFilter
+                    ? execution.executionStatus === selectedFilter
+                    : true
+                )
```

Comment:
pull it above with useMemo to avoid execution on each renders

Reply:
Made the suggested changes